ZenPacks/zenoss/Microsoft/Windows/datasources/ProcessDataSource.py

```diff
@@ -418,29 +418,32 @@ def onSuccess(self, results, config):
             (c, p) for c, p in pids_by_component.iteritems() if p)
 
         # Win32_PerfFormattedData_PerfProc_Process: Datapoints.
-        perf_key = [x for x in results if 'Win32_Perf' in x.wql][0]
-        for item in results[perf_key]:
-            if item.IDProcess not in datasource_by_pid:
-                continue
-
-            datasource = datasource_by_pid[item.IDProcess]
-            for point in datasource.points:
-                if point.id == COUNT_DATAPOINT:
+
+        try:
+            perf_key = [x for x in results if 'Win32_Perf' in x.wql][0]    
+            for item in results[perf_key]:
+                if item.IDProcess not in datasource_by_pid:
                     continue
+                datasource = datasource_by_pid[item.IDProcess]
+                for point in datasource.points:
+                    if point.id == COUNT_DATAPOINT:
+                        continue
 
-                try:
-                    value = int(getattr(item, point.id))
-                except (TypeError, ValueError):
-                    LOG.warn(
-                        "%s %s %s: Couldn't convert %r to integer",
-                        datasource.device, datasource.component, point.id,
-                        value)
-                except AttributeError:
-                    LOG.warn(
-                        "%s %s: %s not in result",
-                        datasource.device, datasource.component, point.id)
-                else:
-                    metrics_by_component[datasource.component][point.id].append(value)
+                    try:
+                        value = int(getattr(item, point.id))
+                    except (TypeError, ValueError):
+                        LOG.warn(
+                            "%s %s %s: Couldn't convert %r to integer",
+                            datasource.device, datasource.component, point.id,
+                            value)
+                    except AttributeError:
+                        LOG.warn(
+                            "%s %s: %s not in result",
+                            datasource.device, datasource.component, point.id)
+                    else:
+                        metrics_by_component[datasource.component][point.id].append(value)
+        except Exception:
```

Comment:
This exception handling is too coarse. What actually causes the IndexError described in ZEN-18823? If it's the creation of `perf_key` using the `[0]` of that list comprehension, maybe we should build that list first and just check that it's len() > 0 before doing the rest.

It's also not clear to me what situation actually causes that list comprehension to return an empty list. Is it really nothing to worry about, or is it a sign that that we were unable to get data? If we were unable to get data we expected to get, simply passing along silently doesn't seem like a good idea.